app/models/email.rb

```diff
@@ -1,5 +1,7 @@
 class Email < ApplicationRecord
   belongs_to :notification
+  has_many :delivery_attempts
+
```

Comment:
I'd be tempted to not add this association to `Email` so that visibility is asymmetric:

- Email doesn't know about DeliveryAttempts
- Delivery attempt knows about Email

It'd mean we'd go from: `email.delivery_attempts` to `DeliveryAttempt.where(email: email)`.

I think the advantage of this approach is it reduces the coupling between these models and encourages us to keep `DeliveryAttempt` specific behaviour out of the `Email` model.

This is probably controversial though? What do you think?

Reply:
Yeah, I actually completely agree with this. I was hesitant to actually include this in the first place...